### ISSUE TITLE
Bitbucket integration not working anymore

### ISSUE THREAD

**Issue by olifly** [nextflow-io/nextflow#1183]
Bitbucket has completely dropped support for their deprecated 1.0 API.  Nextflow still relies on that endpoint for it's bitbucket integration.

### Expected behavior and actual behavior

__Expected:__ Nextflow should be able to run nextflow git repositories hosed on bitbucket

__Actual:__ Nextflow fails when trying to validate and fetch the git repository.

### Steps to reproduce the problem

* Try to run a nextflow script repository hosted on bitbucket.

### Program output 

Relevant errors from the nextflow log file:

```
Jun-12 10:22:31.517 [main] INFO  nextflow.cli.CmdRun - Pulling nextcode-health/nextflow-test ...
Jun-12 10:22:31.526 [main] DEBUG nextflow.scm.RepositoryProvider - Request [credentials <redacted>:************] -> https://bitbucket.org/api/1.0/repositories/nextcode-health/nextflow-test/main-branch/
Jun-12 10:22:31.824 [main] DEBUG nextflow.scm.AssetManager - Project manifest does not exist: https://bitbucket.org/api/1.0/repositories/nextcode-health/nextflow-test/main-branch/
Jun-12 10:22:31.827 [main] DEBUG nextflow.scm.RepositoryProvider - Request [credentials <redacted>:************] -> https://bitbucket.org/api/1.0/repositories/nextcode-health/nextflow-test/main-branch/
Jun-12 10:22:31.844 [main] DEBUG nextflow.scm.RepositoryProvider - Request [credentials <redacted>:************] -> https://bitbucket.org/api/2.0/repositories/nextcode-health/nextflow-test
Jun-12 10:22:32.902 [main] DEBUG nextflow.cli.Launcher - Operation aborted
java.io.FileNotFoundException: https://bitbucket.org/api/1.0/repositories/nextcode-health/nextflow-test/main-branch/
```

### Environment 

* Nextflow version: Version 19.05.0-edge 
* Java version: 1.8.0_181
* Operating system:  macOs

**Comment by pditommaso** [comment]
Doesn't look so, I've managed to use it. 

```
Jun-15 18:44:36.829 [main] DEBUG nextflow.scm.RepositoryProvider - Request [credentials -:-] -> https://bitbucket.org/api/1.0/repositories/pditommaso/tutorial/main-branch/
Jun-15 18:44:37.600 [main] DEBUG nextflow.scm.RepositoryProvider - Request [credentials -:-] -> https://bitbucket.org/api/1.0/repositories/pditommaso/tutorial/src/master/nextflow.config
Jun-15 18:44:37.750 [main] DEBUG nextflow.scm.AssetManager - Project manifest does not exist: https://bitbucket.org/api/1.0/repositories/pditommaso/tutorial/src/master/nextflow.config
Jun-15 18:44:37.759 [main] DEBUG nextflow.scm.RepositoryProvider - Request [credentials -:-] -> https://bitbucket.org/api/1.0/repositories/pditommaso/tutorial/src/master/main.nf
Jun-15 18:44:38.237 [main] DEBUG nextflow.scm.RepositoryProvider - Request [credentials -:-] -> https://bitbucket.org/api/2.0/repositories/pditommaso/tutorial
Jun-15 18:44:38.395 [main] DEBUG nextflow.scm.AssetManager - Pulling pditommaso/tutorial -- Using remote clone url: https://bitbucket.org/pditommaso/tutorial.git
Jun-15 18:44:39.413 [main] INFO  nextflow.cli.CmdRun -  downloaded from https://bitbucket.org/pditommaso/tutorial.git
```

Can you provide more details?

**Comment by olifly** [comment]
@pditommaso 

Hmmm yeah, it seems they must have re-enabled the 1.0 API again.  It's working for me again now

Here's the deprecation notice from Atlassian: https://developer.atlassian.com/cloud/bitbucket/deprecation-notice-v1-apis/

I would recommend taking a look at the pull request I made that moves nextflow to all 2.0 API since it's just a matter of time until Atlassian finally pulls the 1.0 off the air.

**Comment by pditommaso** [comment]
Definitively agree, just wanted to understand the context. Will at the earliest convenience.

**Comment by pditommaso** [comment]
Fixed by #1184 see feeddcda.
